webapp/lib/controller.dart

```diff
@@ -610,6 +619,27 @@ void command(UIAction action, Data data) {
       selectedSuggestedRepliesCategory = updateCategoryData.category;
       _populateReplyPanelView(suggestedRepliesByCategory[selectedSuggestedRepliesCategory]);
       break;
+    case UIAction.hideAgeTags:
+      ToggleData toggleData = data;
+      hideAgeTags = toggleData.toggleValue;
+      switch (actionObjectState) {
+        case UIActionObject.conversation:
```

Comment:
Should age tags be un/hidden in both conversations and message together? Just trying to understand when we want age tags to be hidden in conversations but visible in messages, or vise versa.

Reply:
Yes, we want tags hidden both in conversation and in messages. However, in the UI, only one set will be visible at a time, so depending on which one we're showing (ie. do we have an entire conversation selected or just a single message, which is given by `actionObjectState`), we need to update the panel listing tags on the RHS with either conversation tags or message tags.

The menu for filtering conversations is based on only conversation tags (though maybe in the future this will change). Hence the need to compute `filteredConversationTags` even when showing message tags in the panel.

I opted for a transient way to filter tags, ie. compute filtered messages as needed rather than keeping track of specifically which set is being displayed at any moment, as this seemed easier and wouldn't add any new state to keep track of.

Rereading the code I think it would make it clearer if I just computed both at the top followed by the switch to populate the panel, followed by populating the menu, so I've went ahead and tidied it up like that.

Does this make more sense now? If not, please let me know!

Reply:
@marianamarasoiu Ah! I didn't understand that only one set at a time would be visible in the UI. All good!